datalake/src/main/java/com/sequenceiq/datalake/flow/datalake/scale/handler/DatalakeHorizontalScaleCmRolingResartWaitHandler.java

```diff
@@ -0,0 +1,113 @@
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleFinishedEvent;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+@Component
+public class DatalakeHorizontalScaleCmRolingResartWaitHandler extends EventSenderAwareHandler<DatalakeHorizontalScaleCmRollingRestartRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeHorizontalScaleCmRolingResartWaitHandler.class);
+
+    private static final Long CONVERT_TO_MILLISEC = 1000L;
+
+    @Value("${sdx.datalake.scale.sleeptime-sec:10}")
+    private int sleepTimeInSec;
+
+    @Value("${sdx.datalake.scale.timeout-sec:15}")
+    private int timeoutSec;
+
+    @Inject
+    private ClouderaManagerApiFactory clouderaManagerApiFactory;
+
+    @Inject
+    private ClouderaManagerProxiedClientFactory clouderaManagerProxiedClientFactory;
+
+    protected DatalakeHorizontalScaleCmRolingResartWaitHandler(EventSender eventSender) {
+        super(eventSender);
+    }
+
+    @Override
+    public String selector() {
+        return DATALAKE_HORIZONTAL_SCALE_CM_ROLLING_RESTART_HANDLER.selector();
+    }
+
+    @Override
+    public void accept(Event<DatalakeHorizontalScaleCmRollingRestartRequest> event) {
+        var data = event.getData();
+        var restartClusterArgs = new ApiRollingRestartClusterArgs();
+        restartClusterArgs.setRedeployClientConfiguration(true);
+        var apiClient = clouderaManagerProxiedClientFactory.getProxiedClouderaManagerClient(data.getResourceCrn());
+        var clustersResourceApi = clouderaManagerApiFactory.getClustersResourceApi(apiClient);
+        var commandId = BigDecimal.ZERO;
+        LOGGER.info("CM rolling restart start for {}", data.getResourceName());
+        try {
+            ApiCommand apiCommand = clustersResourceApi.rollingRestart(data.getResourceName(), restartClusterArgs);
+            commandId = apiCommand.getId();
+        } catch (ApiException e) {
+            throw new RuntimeException(e);
+        }
+        LOGGER.debug("Rolling restart command id: {}", commandId.longValue());
+        pollCM(apiClient, commandId);
+        var finishedEvent = new DatalakeHorizontalScaleFinishedEvent(
+                DATALAKE_HORIZONTAL_SCALE_FINISHED_EVENT.selector(),
+                data.getResourceId(),
+                data.getResourceName(),
+                data.getUserId(),
+                commandId
+        );
+        eventSender().sendEvent(finishedEvent, event.getHeaders());
+    }
+
+    private void pollCM(ApiClient apiClient, BigDecimal commandId) {
+        CommandsResourceApi commandsResourceApi = clouderaManagerApiFactory.getCommandsResourceApi(apiClient);
+        Instant timeoutLimit = Instant.now().plusSeconds(timeoutSec);
+        while (isRunning(commandId, commandsResourceApi) && !isTimedout(timeoutLimit)) {
```

Comment:
so if restart timeout occurs we just assume everything is ok and continue the flow?